src/client/src/containers/RightSidebar/index.tsx

```diff
@@ -151,82 +188,104 @@ class RightSidebar extends React.Component<Props, IRightSidebarState> {
     } = this.props;
     const { formatMessage } = intl;
     const { frontendOptions, backendOptions } = contentOptions;
+    const { sideBarOpen } = this.state;
+
+    if (pathname === ROUTES.PAGE_DETAILS || pathname === ROUTES.NEW_PROJECT) {
```

Comment:
I feel like you could use `pathname === ROUTES.PAGE_DETAILS || pathname === ROUTES.NEW_PROJECT` somewhere else to set `sideBarOpen`. Because you are using that logic (`pathname === ROUTES.PAGE_DETAILS || pathname === ROUTES.NEW_PROJECT` ) to also display or not display the sidebar. Then this conditional could just be `if(sideBarOpen) return null;

Reply:
Actually the logic is quite different here.
When !sideBarOpen, we display the hamburger, while on details and new_project we don't render anything.

Reply:
What you're saying makes sense but it's still kind of confusing. There are three states of the side bar. I guess I find it confusing to use `sidebarOpen` but there's a third state in there.

Reply:
I feel like that "should we render it at all" logic should be passed in as a prop `disabled` and then you do the pathname=== logic in the parent.

Reply:
I agree the code may be confusing.
The problem here is that the parent is App component and I don't want to put this logic there.
I'll try with disabled state, but calculated in Sidebar component.

Reply:
At the second thought, I don't see why not put this logic in App component, we already check pathname there for splash background :)